File: library/src/main/java/com/pengrad/telegrambot/AttachName.java
package com.pengrad.telegrambot;

import java.util.concurrent.atomic.AtomicInteger;

/**
 * Stas Parshin
 * 29 July 2018
 */
public class AttachName {

    private AttachName() {}

    private static final AtomicInteger counter = new AtomicInteger();

    public static String next() {
        return "attach" + counter.incrementAndGet();
    }

    public static void reset() {
        counter.set(0);
    }

}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/AttachName.kt
package com.pengrad.telegrambot

import java.util.concurrent.atomic.AtomicInteger

/**
 * Stas Parshin
 * 29 July 2018
 */
object AttachName {

    private val counter = AtomicInteger()

    operator fun next(): String {
        return "attach" + counter.incrementAndGet()
    }

}
